src/server/game/Entities/Object/Object.cpp

```diff
@@ -1520,9 +1520,7 @@ float WorldObject::GetGridActivationRange() const
 
 float WorldObject::GetVisibilityRange() const
 { 
-    if (isActiveObject() && !ToPlayer())
```

Comment:
why you removed this check?

Reply:
No need to increase visibility distance for active creatures. This should be controlled solely via "isLarge" flag in creature_addon or creature_template_addon.